src/dbt_score/formatters/json_formatter.py

```diff
@@ -0,0 +1,100 @@
+"""JSON formatter.
+
+Shape of the JSON output:
+
+{
+    "models": {
+        "model_foo": {
+            "score": 0.5,
+            "badge": "🥈",
+            "results": {
+                "rule1": {
+                    "result": "OK",
+                    "severity": null
+                    "message": null
+                },
+                "rule2": {
+                    "result": "WARN",
+                    "severity": "medium",
+                    "message": "Model lacks a description."
+                }
+            ]
+        },
+        "model_bar": {
+            "score": 0.0,
+            "badge": "🥉",
+            "results": {
+                "rule1": {
+                    "result": "ERR",
+                    "message": "Exception message"
+                }
+            }
+        }
+    },
+    "project": {
+        "score": 0.25,
+        "badge": "🥉"
+    }
+}
+"""
+
+
+import json
+from typing import Any
+
+from dbt_score.evaluation import ModelResultsType
+from dbt_score.formatters import Formatter
+from dbt_score.models import Model
+from dbt_score.rule import RuleViolation
+from dbt_score.scoring import Score
+
+
+class JSONFormatter(Formatter):
+    """Formatter for JSON output."""
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        """Instantiate formatter."""
+        super().__init__(*args, **kwargs)
+        self._model_results: dict[str, dict[str, Any]] = {}
+        self._project_results: dict[str, Any]
+
+    def model_evaluated(
+        self, model: Model, results: ModelResultsType, score: Score
+    ) -> None:
+        """Callback when a model has been evaluated."""
+        self._model_results[model.name] = {
+            "score": score.value,
+            "badge": score.badge,
+            "results": {},
+        }
+        for rule, result in results.items():
+            if result is None:
+                self._model_results[model.name]["results"][rule.source()] = {
+                    "result": "OK",
```

Comment:
Maybe we can make this an Enum or something, so it's clear what are the possible results codes?

Reply:
That's a good idea, but given it's also used in the `HumanReadableFormatter`, this would require a bit of refactoring to have it proper. I suggest to keep it simple for now :)